var/spack/repos/builtin/packages/mumps/package.py

```diff
@@ -174,11 +199,20 @@ def write_makefile_inc(self):
                     'RANLIB=echo'
                 ])
             else:
-                makefile_conf.extend([
-                    'LIBEXT=.so',
-                    'AR=$(FL) -shared -Wl,-soname -Wl,%s/$(notdir $@) -o' % prefix.lib,  # noqa
-                    'RANLIB=echo'
-                ])
+                if self.compiler.name == 'xl' or self.compiler.name == 'xl_r':
+                    makefile_conf.extend([
+                        'LIBEXT=.so',
+                        'AR=$(FL) -shared -Wl,-soname -Wl,%s/$(notdir $@) -o' % prefix.lib,  # noqa
+                        'SAR=/bin/xlc -shared -Wl,-soname -Wl,%s/$(notdir $@) -o' % prefix.lib,  # noqa
```

Comment:
that's the only new line, no need to copy-paste everything, just add this extra line in additional `if`.